sdv/constraints/base.py

```diff
@@ -236,6 +262,8 @@ def _validate_constraint_columns(self, table_data):
             table_data (pandas.DataFrame):
                 Table data.
         """
+        self._validate_data_on_constraint(table_data)
```

Comment:
These two methods: `_validate_data_on_constraint` and `_validate_constraint_columns` should be merged in the future, when the `fit_columns_model` logic gets dropped.

Reply:
Let's keep these separate for a couple of reasons:
1. We may end up keeping the columns model
2. These are actually two different types of validation. One is checking that the data being transformed actually adheres to the constraint. The other is checking if any columns in the constraint are missing and need to be sampled. Let's keep the logic separate. Maybe changing the name of `_validate_constraint_columns` to `_generate_missing_columns` would help or something like that